File: Algorithm/Other/at coder/Other/atcoder/A40/A40.py
from collections import Counter
import sys

def main():
    # 入力の読み取り
    N = int(sys.stdin.readline())
    A = list(map(int, sys.stdin.readline().split()))

    # 長さごとの本数をカウント
    count = Counter(A)

    result = 0
    for c in count.values():
        if c >= 3:
            result += c * (c - 1) * (c - 2) // 6  # C(c, 3)

    print(result)

if __name__ == "__main__":
    main()

File: Algorithm/Other/at coder/Other/A40/A40.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin', 'utf8');

function countEquilateralTriangles(input) {
    const lines = input.trim().split('\n');
    const N = parseInt(lines[0], 10);
    const A = lines[1].split(' ').map(Number);

    // 長さごとにカウント（1 <= Ai <= 100）
    const count = Array(101).fill(0);
    for (let i = 0; i < N; i++) {
        count[A[i]]++;
    }

    // 長さごとの3本の組み合わせを計算
    let result = 0;
    for (let i = 1; i <= 100; i++) {
        if (count[i] >= 3) {
            const c = count[i];
            result += (c * (c - 1) * (c - 2)) / 6; // 組み合わせC(n,3)
        }
    }

    console.log(result);
}

countEquilateralTriangles(input);
